packages/proxy/src/proxy.ts

```diff
@@ -309,55 +331,64 @@ export async function proxyV1({
     const cached = await cacheGet(encryptionKey, cacheKey);
 
     if (cached !== null) {
-      cacheHits.add(1);
       const cachedData: CachedData = JSON.parse(cached);
+      // XXX simplify once all cached data has a timestamp - assume existing data has age of 7 days
+      const age = cachedData.timestamp
+        ? Math.floor(getCurrentUnixTimestamp() - cachedData.timestamp)
+        : DEFAULT_CACHE_TTL;
+
+      if (!cacheMaxAge || age <= cacheMaxAge) {
+        cacheHits.add(1);
+        for (const [name, value] of Object.entries(cachedData.headers)) {
+          setHeader(name, value);
+        }
+        setHeader(CACHED_HEADER, "HIT");
+        setHeader("cache-control", `max-age=${cacheTTL}`);
```

Comment:
should we give the ttl that the user gave here, or how long the entry has left to live?

Reply:
yeah this should be the requested ttl. ill add to the cached metadata and update